### DISCUSSION TITLE
First DRAFT release

### DISCUSSION THREAD

**Discussion by pschatzmann** [pschatzmann/arduino-snapclient#5] [Announcements]
First DRAFT release with
- support for Opus
- start time synchronization

<hr /><em>This discussion was created from the release <a href='https://github.com/pschatzmann/arduino-snapclient/releases/tag/v0.1.0'>First DRAFT release</a>.</em>

**Comment by CarlosDerSeher** [comment]
@pschatzmann I've been looking at your code but couldn't really pinpoint it. How does your client synchronization work specifically? 

I use apll tuning over at my IDF implementation but it seems there are possible issues with some DAC using this approach. I also did try to insert/remove samples but couldn't make it work properly.

So where in the playback path do you tune the playback speed to keep up sync over time?

**Comment by pschatzmann** [comment]
Did you read the [Wiki](https://github.com/pschatzmann/arduino-snapclient/wiki) ?
The currently provided implementations can be found in the SnapTimeSync.h.
It relies on the [resampling](https://github.com/pschatzmann/arduino-audio-tools/wiki/Resampling) functionality provided by the AudioTools...

**Reply by CarlosDerSeher** [reply]
> Did you read the Wiki ?

Seems I've somehow missed that. Thanks for the hint.

> It relies on the resampling functionality provided by the AudioTools...

So you set the sample rate sent by snapserver and upsample / downsample your pcm buffer as needed? You do that on a wire chunk basis? Could you quantify on how good the sync is, e.g. 100us deviation max or so?
